src/client/eppo-client.ts

```diff
@@ -651,16 +652,39 @@ export default class EppoClient {
   }
 
   private logBanditAction(banditEvent: IBanditEvent): void {
+    const subjectKey = banditEvent.subject;
+    const flagKey = banditEvent.featureFlag;
+    const banditKey = banditEvent.bandit;
+    const actionKey = banditEvent.action ?? '__eppo_no_action';
+
+    // What our bandit assignment cache cares about for avoiding logging duplicate bandit assignments,
+    // if one is active. Like the flag assignment cache, entries are only stored for a given flag
+    // and subject. However, Bandit and action keys are also used for determining assignment uniqueness.
+    // This means that if a flag's bandit assigns one action, and then later a new action, the first
+    // one will be evicted from the cache. If later assigned again, it will be treated as new.
+    const banditAssignmentCacheProperties = {
+      flagKey,
+      subjectKey,
+      banditKey,
+      actionKey,
+    };
+
+    if (this.banditAssignmentCache?.has(banditAssignmentCacheProperties)) {
+      // Ignore repeat assignment
+      return;
+    }
+
     if (!this.banditLogger) {
       // No bandit logger set; enqueue the event in case a logger is later set
       if (this.queuedBanditEvents.length < MAX_EVENT_QUEUE_SIZE) {
         this.queuedBanditEvents.push(banditEvent);
```

Comment:
Q: shall we deduplicate events in the `queuedBanditEvents`?

Reply:
I don't think so as these are basically all sent to a logger if one is added, so we want deduping to happen beforehand like we're doing

Reply:
We're not deduplicating them beforehand here because `return` on the next line prevents us from setting the entry in the cache.

If we log the same event twice before the user sets the logger, the event will be sent to user's logger twice

Reply:
Ah I see what you are saying now, good catch! Will address